Gitlab CICD.sublime-syntax

```diff
@@ -63,3 +63,40 @@ contexts:
     - meta_include_prototype: false
     - meta_scope: source.shell.bash.embedded
     - include: scope:source.shell.bash.folded
+
+  property:
+    - meta_prepend: true
+    - match: (!reference)\s*(\[)
+      captures:
+        1: meta.property.yaml storage.type.tag-handle.yaml
+        2: meta.sequence.flow.yaml punctuation.definition.sequence.begin.yaml
+      push: gitlab-reference-flow-sequence-body
+
+  gitlab-reference-flow-sequence-body:
+    - meta_content_scope: meta.sequence.flow.yaml
+    - match: \]
+      scope: meta.sequence.flow.yaml punctuation.definition.sequence.end.yaml
+      pop: 1
+    - match: ','
+      scope: punctuation.separator.sequence.yaml
+      push: gitlab-after-reference
+    - match: '[-\w.]+'
+      scope: constant.other.label.gitlab
+    - include: flow-pair-no-clear
+    - include: flow-node-11
+
+  gitlab-after-reference:
+    - match: \w+
+      scope: keyword.other.gitlab
+    - match: ','
+      scope: punctuation.separator.sequence.yaml
+      set: gitlab-after-reference-type
+    - match: (?=\])
+      pop: 1
```

Comment:
Missing flow-node-11 (or 12?) include for quoted scalars.

Reply:
I wonder if we can (in the future) use that multiple-inheritance technique to make some of this a lot easier - like we want to scope the text a certain way regardless of whether it is unquoted, single quoted or double quoted... Otherwise it is a lot of work / custom rules...

Reply:
All segments of the `!reference` sequence could simply have string scopes imo. If it becomes too complicated, we should look into making compromises.